### DISCUSSION TITLE
MQTT dopo la versione 2022.12.0 home assistant

### DISCUSSION THREAD

**Discussion by Batteralex** [indomus/forum#1905] [HUB: Home Assistant]
Salve a tutti, da ieri mi compaiono questi messaggi sul mio home assistant riferiti a più elementi (switch, light, cover e sensor) che prevedeono a quanto pare una cessione della funzionalità...

"La/e tua/e cover MQTT configurata/e manualmente ha/hanno bisogno di attenzione
Questo smette di funzionare nella versione 2022.12.0. Risolvi il problema prima di effettuare l'aggiornamento.
cover MQTT configurata manualmente trovata sotto la voce della piattaforma cover.

Sposta la configurazione sulla voce di integrazione mqtt e riavvia Home Assistant per risolvere questo problema. Consulta la [documentazione](https://www.home-assistant.io/integrations/cover.mqtt/#new_format), per ulteriori informazioni."

Cosa e come bisognerebbe fare per sistemare tutto?
Grazie

**Comment by snakuzzo** [comment]
E' tutto scritto sia nella [doc ufficiale](https://www.home-assistant.io/integrations/mqtt/) che nella [nostra news](https://indomus.it/news/hot-aggiornamento-2022-6-per-home-assistant-core/) della release 2022.06

**Reply by sky7176** [reply]
Ciao,
io sono 4 giorni che cerco di capire dove ho questi due 
- sensor MQTT configurata manualmente trovata sotto la voce della piattaforma sensor
- switch MQTT configurata manualmente trovata sotto la voce della piattaforma switch

Gli altri gli ho spostati, ma questi due proprio non li trovo nel file di configurazione "configuration.yaml".
Sbaglio sicuramente io da qualche parte, ma non ho idea di dove siano.

Giuseppe

**Reply by snakuzzo** [reply]
Non è che per caso usi i file sidecar per sensori e switch ?
Nel caso, probabile che siano in quei due file.

**Reply by sky7176** [reply]
Perdona la mia completa ignoranza ma cosa sono i file sidecar?

Giuseppe

**Reply by sky7176** [reply]
Aggiungo anche una cosa, se faccio una verifica della configurazione, il sistema mi dice che è tutto ok, solo la funzione "Riparazioni" mi segnala le due anomalie.
Al contrario, prima di sistemare tutti gli altri aspetti di MQTT, il sistema non mi faceva riavviare perche indicava anomalie nel file di configurazione

**Reply by snakuzzo** [reply]
Non ti segnala anomalie perché questa configurazione puoi usarla fino a fine anno ( mi sembra ).
Per i file "sidecar" ... leggi [QUI](https://indomus.it/guide/configuration-yaml-capire-il-file-di-configurazione-di-home-assistant/)

Nel `configuration.yaml` avrai qualcosa del genere...

```yaml
light: !include light.yaml
sensor: !include sensor.yaml
switch: !include switch.yaml
```

**Reply by sky7176** [reply]
Grazie,
capito anche i file sidecar, ma non sono nemmeno quelli che creano il casino.

Il tutto dovrebbe bloccarsi con la versione 2022.12, ma :
1) mi da fastidio vedere quei due avvisi
2) mi da ancora piùfastidio non capire dove sia il casino

**Reply by sky7176** [reply]
Aggiornamento.
Uno dei due errori, quello relativo allo switch è andato via.
Quello che ho fatto è stato aggiornare delle palette di Node-Red, ma prima di fare questo avevo cambiato la lingua da italiano ad inglese per poter fare gli screenshots, da quel momento mi sono accorto che uno dei due avvisi era scomparo.
Ho ricambiato la lingua ed ho riavviato Home Assistant, ma è rimasto sol un avviso, quello relativo a "sensor".
Ho scritto sul forum di Home-assistantat per chiedere info, non credo che siano stati gli aggiornamenti delle palette di Node-Red, ma credo più il cambio lingua.

Ah se pesco quello che ha detto "l'informatica è una scienza esatta" HI

Giuseppe

**Reply by snakuzzo** [reply]
Ma cosa c'entra il cambio lingua ?????

**Reply by snakuzzo** [reply]
Vabbè manco ti rispondo sull'ultima frase :D

**Reply by sky7176** [reply]
> Ma cosa c'entra il cambio lingua ?????

Nel senso che la segnalazione relativa a "switch" è scomparsa da sola, io non ho modificato nessun file, dopo che ho cambiato la lingua da italiano ad inglese per fare gli screenshots perchè ho aperto una segnalazione sul forum di Home-Assistant.
L'altra cosa che ho fatto è stata quella di aggiornare delle palette di Node-Red, ma non credo che questo centri.
Ora, vero che ho tempo sino a dicembre, ma se non trovo il bandolo della matassa......

Giuseppe

**Reply by Scialla** [reply]
Se non trovi il bandolo della matassa smetterà di funzionare l'entità legata all'errore e non avrai più dubbi su cosa cercare :)

**Reply by sky7176** [reply]
> Se non trovi il bandolo della matassa smetterà di funzionare l'entità legata all'errore e non avrai più dubbi su cosa cercare :)

Hai ragione ..... HI

**Reply by sky7176** [reply]
Risolto,
era il file zigbee2mqtt.yaml contenuto nella cartella packages che faceva casino.
Mi era sfuggito perchè non era un file creato da me.

Snakuzzo, avevi ragione era un file Sidecar, ma sarebbe bello che le segnalazioni di anomalie indicassero dove andare a cercare, il sistema inizia ad essere molto complesso.

 Giuseppe

**Reply by snakuzzo** [reply]
Beh...complesso non direi proprio e non capisco di che file parli...non mi pare che zigbee2mqtt crei file per conto suo (almeno su core)

**Reply by sky7176** [reply]
Il file è contenuto nella cartella packages in config, la data di creazione è di quando ho installato l'integrazione zigbee2mqtt nel 2020.
Giuseppe

**Comment by Batteralex** [comment]
Anche io ho i file sidecar per quanto riguarda switch sensor light e cover ma non riesco in nessun modo a farlo funzionare.
Seguo alla lettera la modifica riportata nell'esempio della documentazione relativa a mqtt mettendo all'interno del file (ad esempio sensor.yaml) l'intestazione mqtt: seguita da sensor: ed eliminando la voce platform: mqtt ma mi rilascia sempre errore.
Sicuramente starà sbagliando qualcosa di banale ma non riesco a capire cosa.
Qualcuno mi può aiutare per favore.
Grazie

**Reply by snakuzzo** [reply]
Premesso che sicuramente stai sbagliando qualcosa, ti vorrei fare ragionare su una coaa...
Immagina una persona che legge il tuo problema così come lo hai esposto. Secondo te come fa a dirti cos'hai sbagliato? Può andarsene solo per un'idea ipotizzando un tuo errore, ma si perderebbe solo tanto tempo.

Quando avete un problema siete pregati di segnalare:

- errori specifici nel file di log
- configurazione (YAML) indentata per bene **usando il markdown** !

P.S. al posto dei sidecar ti consiglierei di valutare l'utilizzo dei package

**Comment by Batteralex** [comment]
Diciamo che riservo molta fiducia nelle persone che leggeranno ciò che scritto certo che capiranno anche se ho scritto male…ma comunque hai pienamente ragione. Ho solo scritto velocemente perché avevo poco tempo e a questo aggiungo il fatto non so usare il markdown.😢

**Reply by snakuzzo** [reply]
Eppure è scritto sull'indice del forum e messo in evidenza

[QUI](https://github.com/indomus/forum/discussions/1311)

**Comment by Batteralex** [comment]
Perfetto. Grazie per la dritta. Con calma riscriverò tutto come si deve.

**Comment by Batteralex** [comment]
Allora, riscrivo il problema in maniera corretta...
nel mio file sidecar sensor.yaml ho cambiato la compilazione da cosi

   ```

    - platform: mqtt
      name: "Temperatura Camera Da Letto"
      state_topic: "tele/luce_corridoio_2/SENSOR" 
      unit_of_measurement: "°C"
      qos: 1
      value_template: '{{ value_json.AM2301.Temperature }}'
   
 ```
a cosi...

```

mqtt:
  sensor:
    - name: "Temperatura Camera Da Letto"
      state_topic: "tele/luce_corridoio_2/SENSOR" 
      unit_of_measurement: "°C"
      qos: 1
      value_template: '{{ value_json.AM2301.Temperature }}'

```
ricevendo questo errore di errata configurazione duranta la verifica:
"Error loading /config/configuration.yaml: expected '<document start>', but found '<block mapping start>'
in "/config/sensor.yaml", line 43, column 1"

e questo nel file sensor al momento della compilazione:
"end of the stream or a document separator is expected (43:1)

 40 | #     T E M P E R A T U R A   S ...
 41 | 
 42 | 
 43 | mqtt:
------^
 44 |   sensor:
 45 |     - name: "Temperatura Camera ...".

Cosa sto sbagliando?

**Reply by Ale1975-Iron** [reply]
Ciao, ma hai riscritto tutto all'interno del tuo file sidecar? credo il problema sia quello... quanto hai scritto va posizionato in configuration.yaml... oppure crei un sidecar x mqtt.

Ciao

**Reply by Batteralex** [reply]
Perfetto. Era proprio quello il problema. Finalmente ho risolto creando un file sidecar mqtt e ricopiando tutto dentro. 
grazie mille per il supporto.
